src/demProfiler/index.ts

```diff
@@ -0,0 +1,237 @@
+import tileCover from "./tileCover";
+import getElevation from "./getElevation";
+import { pointDistance, mToFt } from "../geometry";
+
+import type { TileCoverCoordinates, TileID } from "./tileCover";
+import type { ElevationParser } from "./getElevation";
+import type { Feature, LineString } from "geojson";
+
+export * from "./getElevation";
+export * from "./tileCover";
+
+export interface TileImage {
+  channels: 1 | 2 | 3 | 4;
+  image: Uint8ClampedArray;
+}
+
+export type TileRequest = (
+  x: number,
+  y: number,
+  zoom: number
+) => Promise<TileImage>;
+
+export interface Options {
+  /** type of metric to use. Meters or feet. Default: meters */
+  metric?: "m" | "ft";
+  /** Zoom that is queried from the server. Default: 13 */
+  zoom?: number;
+  /** Tile size of the images returned. Default 512 */
+  tileSize?: number;
+  /** Tile Request method */
+  tileRequest: TileRequest;
```

Comment:
I mentioned that in another comment somewhere else, but having a default tile request (just the request, not the parsing) that already goes to fetch MapTiler Terrain data, concatenated with the apiKey in the config object would be super nice. (then all the options are actually optional)

Reply:
Unfortunately this library supports UMD, so without treeshake, whichever one I support will break the other. Node uses native libraries like sharp, and browser.

Lastly, writing this code is very simple and takes only a few minutes. If at some point this library supports documentation here are the two types that can be written down to be copy pasted:

```ts
import sharp from 'sharp'

const options = {
  metric: 'm' as 'm' | 'ft',
  zoom: 13,
  tileSize: 512,
  tileRequest: async (x: number, y: number, z: number) => {
  const buf = await sharp(`test/fixtures/${z}-${x}-${y}.webp`)
      .raw()
      .toBuffer({ resolveWithObject: true })
    return {
      channels: buf.info.channels,
      image: new Uint8ClampedArray(buf.data)
    }
  },
 }
```

browser:

```ts
async function tileRequest(
  x: number,
  y: number,
  zoom: number,
  apiKey: string
): Promise<TileImage> {
  const url = `https://www.tileAPI.com/${zoom}/${x}/${y}.png?key=${apiKey}`;
  const imageData = await requestAndGetImageData(url).catch((err) => {
    throw new Error(`Could not load image from ${url}.\n${err}`);
  });

  return {
    channels: 4,
    image: imageData.data,
  };
}

async function requestAndGetImageData(url: string): Promise<ImageData> {
  const res = await fetch(url);
  const image = await res.blob().catch((err) => {
    return err;
  });
  const bitmap = await createImageBitmap(image);
  const canvas = document.createElement("canvas");
  canvas.width = bitmap.width;
  canvas.height = bitmap.height;
  const context = canvas.getContext("2d");
  if (context === null) throw new Error("Could not create a context");
  context.drawImage(bitmap, 0, 0);
  return context.getImageData(0, 0, bitmap.width, bitmap.height);
}

```

```